### PR TITLE
Allow search to be toggled on/off from the UI

### PR DESCRIPTION
**Pull request by chuyskywalker**
See: #1586, #1795, #2722

Cleaned up, no globals, doesn't "restart" threads, just pauses the overall process; handles relocated marker during "paused" state.

### CODE REVIEW

**Comment by Chlodochar** [comment]
Tested and working. This is a very welcome addition. Code looks good, except that Landscape.io found one unused parameter, could you remove that one?

**Comment by mackhankins** [comment]
Very clean.  I tested the following on my mac.  I'm going to push it out to Heroku for some "field testing"

* Follow Location with Search on & off
* Drag Marker with Search on & off

@chuyskywalker Holy branches 👍

File: pogom/app.py [hunk 1]
```diff
@@ -28,19 +28,39 @@ def __init__(self, import_name, **kwargs):
...
+        self.route("/search_control", methods=['POST'])(self.post_search_control)
+
+    def set_search_control(self, control):
+        self.search_control = control
+
+    def get_search_control(self):
+        return jsonify({'status': self.search_control.is_set()})
+
+    def post_search_control(self):
+        action = request.args.get('action','none')
```
Comment (chuyskywalker) [inline pogom/app.py]: I just realized I've missed something here wherein this endpoint is still available even when the flag `-nsc` is turned on. Need to detect that flag and disable this endpoint.

Comment (chuyskywalker) [inline pogom/app.py]: Fixed

**Comment by magimat** [comment]
been testing this feature tonight, works great

**Comment by medghaim** [comment]
+1

**Comment by mackhankins** [comment]
Thanks :)

**Comment by magimat** [comment]
hmm.. I'm running the develop branch including this PR since this morning and it seems each time I resume the search after pausing for a long while (maybe an hour), it doesn't resume correctly. I'm getting the "Niantic server under heavy load" message after resuming and the only way to make it search again is to kill the server and start it back up manually.

**Comment by chuyskywalker** [comment]
Ah dang, I think I see the issue here. Since this just "pauses" all the threads, the login token can will go invalid over time. And when we "unpause" it doesn't validate or re-login, so it gets stuck trying to use the same broken token in an infinite loop. (This is probably even happening _regardless_ of this feature, like if the token expiration rolls around in the middle of a search loop.)

I'll look into a fix for this tonight -- it'll probably require a lot of logical re-writing though :|

**Comment by magimat** [comment]
thanks!  I'd help, but I don't really have the skills nor time to really be helpful other than reporting some test feedback.

thanks for your work.. it's really appreciated!

**Comment by invisiblek** [comment]
I notice search_control gets called, even when I'm not dicking around with the toggle. Might be worth looking in to, seems to be unnecessary calls to this going on.

**Comment by JamesWP** [comment]
If this is the same as my initial version it is called to keep the toggle In sync with the backend.
I.e. if you toggle on one page the other is also updated


Sent from my Samsung device

-------- Original message --------

**Comment by invisiblek** [comment]
Hmm, yea I didn't think of that. I suppose that makes sense. Seems a bit costly to me though...

**Comment by chuyskywalker** [comment]
You're not wrong, but we've got a polling architecture, so it's what we have to work with.

Websockets now... ;D

**Pull request merged**